omnigibson/utils/usd_utils.py

```diff
@@ -193,58 +197,37 @@ def create_joint(prim_path, joint_type, body0=None, body1=None, enabled=True, st
     og.sim.render()
 
-    # Return this joint
-    return joint_prim
-
-
-class CollisionAPI:
-    """
-    Class containing class methods to facilitate collision handling, e.g. collision groups
-    """
-    ACTIVE_COLLISION_GROUPS = {}
-
-    @classmethod
-    def add_to_collision_group(cls, col_group, prim_path, create_if_not_exist=False):
-        """
-        Adds the prim and all nested prims specified by @prim_path to the global collision group @col_group. If @col_group
-        does not exist, then it will either be created if @create_if_not_exist is True, otherwise will raise an Error.
+    if joint_frame_in_parent_frame_pos is not None:
+        joint_prim.GetAttribute("physics:localPos0").Set(Gf.Vec3f(*joint_frame_in_parent_frame_pos))
+    if joint_frame_in_parent_frame_quat is not None:
+        joint_frame_in_parent_frame_quat = joint_frame_in_parent_frame_quat[[3, 0, 1, 2]]
+        joint_prim.GetAttribute("physics:localRot0").Set(Gf.Quatf(*joint_frame_in_parent_frame_quat))
+    if joint_frame_in_child_frame_pos is not None:
+        joint_prim.GetAttribute("physics:localPos1").Set(Gf.Vec3f(*joint_frame_in_child_frame_pos))
+    if joint_frame_in_child_frame_quat is not None:
+        joint_frame_in_child_frame_quat = joint_frame_in_child_frame_quat[[3, 0, 1, 2]]
+        joint_prim.GetAttribute("physics:localRot1").Set(Gf.Quatf(*joint_frame_in_child_frame_quat))
+
+    if break_force is not None:
+        joint_prim.GetAttribute("physics:breakForce").Set(break_force)
+    if break_torque is not None:
+        joint_prim.GetAttribute("physics:breakTorque").Set(break_torque)
 
-        Args:
-            col_group (str): Name of the collision group to assign the prim at @prim_path to
-            prim_path (str): Prim (and all nested prims) to assign to this @col_group
-            create_if_not_exist (bool): True if @col_group should be created if it does not already exist, otherwise an
-                error will be raised
-        """
-        # TODO: This slows things down and / or crashes the sim with large number of objects. Skipping this for now, look into this later
-        pass
-        # # Check if collision group exists or not
-        # if col_group not in cls.ACTIVE_COLLISION_GROUPS:
-        #     # Raise error if we don't explicitly want to create a new group
-        #     if not create_if_not_exist:
-        #         raise ValueError(f"Collision group {col_group} not found in current registry, and create_if_not_exist"
-        #                          f"was set to False!")
-        #     # Otherwise, create the new group
-        #     col_group_name = f"/World/collisionGroup_{col_group}"
-        #     group = UsdPhysics.CollisionGroup.Define(get_current_stage(), col_group_name)
-        #     group.GetFilteredGroupsRel().AddTarget(col_group_name)  # Make sure that we can collide within our own group
-        #     cls.ACTIVE_COLLISION_GROUPS[col_group] = group
```

Comment:
Oh I see the CollisionAPI was removed altogether lol...we should check to see if this API has been fixed from the omni side.

Reply:
unless it's unnecessary? maybe we don't need it after all

Reply:
added back